src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -398,11 +398,20 @@ private static LibraryIncludeFlags GetIncludeFlags(string value, LibraryIncludeF
             }
         }
 
-        private static string[] Split(string s)
+        /// <summary>
+        /// Split on ; and trim. Null or empty inputs will return an 
+        /// empty array.
+        /// </summary>
+        public static string[] Split(string s)
         {
             if (!string.IsNullOrEmpty(s))
             {
-                return s.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+                // Split on ; and trim all entries
+                // After trimming remove any entries that are now empty due to trim.
+                return s.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(entry => entry.Trim())
+                    .Where(entry => !string.IsNullOrEmpty(entry))
```

Comment:
Just looked at the implementation
```c#
public static bool IsNullOrEmpty(string value)
{
	return value == null || value.Length == 0;
}
```
So the where clause can be "simplified" to `entry.Length !=0`